From d74e6a9e51ff153e227a4230df57763eea0a87ce Mon Sep 17 00:00:00 2001
From: Vladimir Oltean <olteanv@gmail.com>
Date: Thu, 21 Nov 2019 22:25:51 +0200
Subject: [PATCH buildroot 16/18] net: dsa: sja1105: Add error message to mgmt
 xmit code path

Make it clear if something has gone wrong, don't just silently drop
packets. This is an abnormal situation.

Signed-off-by: Vladimir Oltean <olteanv@gmail.com>
---
 drivers/net/dsa/sja1105/sja1105_main.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/drivers/net/dsa/sja1105/sja1105_main.c b/drivers/net/dsa/sja1105/sja1105_main.c
index 73649c99d5a5..e50700073554 100644
--- a/drivers/net/dsa/sja1105/sja1105_main.c
+++ b/drivers/net/dsa/sja1105/sja1105_main.c
@@ -1878,6 +1878,8 @@ static int sja1105_mgmt_xmit(struct dsa_switch *ds, int port, int slot,
 	rc = sja1105_dynamic_config_write(priv, BLK_IDX_MGMT_ROUTE,
 					  slot, &mgmt_route, true);
 	if (rc < 0) {
+		dev_err_ratelimited(priv->ds->dev,
+				    "failed to program mgmt route\n");
 		kfree_skb(skb);
 		return rc;
 	}
-- 
2.17.1

